portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -117,14 +138,39 @@ private int getNumberOfComments(HttpServletRequest request){
     }
 
     return parameter;
+
   }
 
   private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+
     String value = request.getParameter(name);
     if (value == null) {
       return defaultValue;
     }
     return value;
+
+  }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
```

Comment:
You're already creating one of these above on line 48, so you should just use that one.